### PR TITLE
fix: #396 - allow user to cancel a model download

### PR DESCRIPTION
**Pull request by louis-jan**
## Problem
- According to #410, user now cannot cancel a downloading process, there are a few cases could lead to a pending forever download.

## Solution
- Downloading button will be Cancel now, app manages requests and perform pause/cancel/abort accordingly (via CTA)

Screenshot 2023-11-02 at 15 37 38
Screenshot 2023-11-02 at 15 55 27

Action items: 

- [ ] Pause & Resume CTA are not available yet, please help on this @urmauur
- [ ] Cloud native cancel action is not defined yet

### CODE REVIEW

File: web/app/_components/ConfirmationModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import { PrimitiveAtom, useAtom } from 'jotai'
+
+interface Props {
+  atom: PrimitiveAtom<boolean>
+  title: string
+  description: string
+  onConfirm: () => void
+}
+
+const ConfirmationModal: React.FC<Props> = ({ atom, title, description, onConfirm }) => {
```
Comment (louis-jan) [inline web/app/_components/ConfirmationModal/index.tsx:13]: This is general ConfirmationModal, @urmauur @namchuai. Don't have to create every modal for each action (delete conversation / model), let just make use of this one. Please help remove other modals @urmauur

**Review by urmauur — APPROVED** [review]

File: web/app/_components/ExploreModelItemHeader/index.tsx [hunk 2]
```diff
@@ -70,17 +75,30 @@ const ExploreModelItemHeader: React.FC<Props> = ({
...
       </Button>
     )
   }
 
+  let cancelDownloadModal =
+    downloadState != null ? (
+      <ConfirmationModal
+        atom={showingCancelDownloadModalAtom}
+        title="Cancel Download"
+        description={`Are you sure you can to cancel the download of ${downloadState?.fileName}`}
```
Comment (louis-jan) [inline web/app/_components/ExploreModelItemHeader/index.tsx:93]: typo

Comment (louis-jan) [reply web/app/_components/ExploreModelItemHeader/index.tsx]: fixed

File: web/app/_components/ConfirmationModal/index.tsx (hunk 1 above)
Comment (urmauur) [reply web/app/_components/ConfirmationModal/index.tsx:13]: @louis-jan yes will do on refactor, i'll be create a modal component without defined wrapper, thank you so much for confirming

**Pull request merged**